crates/matrix-sdk-sqlite/src/event_cache_store.rs

```diff
@@ -1948,6 +1948,172 @@ mod tests {
+                            items: vec![event("saucisse de morteau"), event("comté")],
+                        },
+                    ],
+                )
+                .await
+                .unwrap();
+
+            let (last_chunk, chunk_identifier_generator) =
+                store.load_last_chunk(room_id).await.unwrap();
+
+            assert_matches!(last_chunk, Some(last_chunk) => {
+                assert_eq!(last_chunk.identifier, 42);
+                assert!(last_chunk.previous.is_none());
+                assert!(last_chunk.next.is_none());
+                assert_matches!(last_chunk.content, ChunkContent::Items(items) => {
+                    assert_eq!(items.len(), 2);
+                    check_test_event(&items[0], "saucisse de morteau");
+                    check_test_event(&items[1], "comté");
+                });
+            });
+            assert_eq!(chunk_identifier_generator.current(), 42);
+        }
+
+        // Case #3: more chunks are present.
+        {
+            store
+                .handle_linked_chunk_updates(
+                    room_id,
+                    vec![
+                        Update::NewItemsChunk {
+                            previous: Some(ChunkIdentifier::new(42)),
+                            new: ChunkIdentifier::new(7),
+                            next: None,
+                        },
+                        Update::PushItems {
+                            at: Position::new(ChunkIdentifier::new(7), 0),
+                            items: vec![event("fondue"), event("gruyère"), event("mont d'or")],
+                        },
+                    ],
+                )
+                .await
+                .unwrap();
+
+            let (last_chunk, chunk_identifier_generator) =
+                store.load_last_chunk(room_id).await.unwrap();
+
+            assert_matches!(last_chunk, Some(last_chunk) => {
+                assert_eq!(last_chunk.identifier, 7);
+                assert_matches!(last_chunk.previous, Some(previous) => {
+                    assert_eq!(previous, 42);
+                });
+                assert!(last_chunk.next.is_none());
+                assert_matches!(last_chunk.content, ChunkContent::Items(items) => {
+                    assert_eq!(items.len(), 3);
+                    check_test_event(&items[0], "fondue");
+                    check_test_event(&items[1], "gruyère");
+                    check_test_event(&items[2], "mont d'or");
+                });
+            });
+            assert_eq!(chunk_identifier_generator.current(), 42);
+        }
+    }
+
+    #[async_test]
+    async fn test_load_previous_chunk() {
+        let room_id = room_id!("!r0:matrix.org");
+        let event = |msg: &str| make_test_event(room_id, msg);
+        let store = get_event_cache_store().await.expect("creating cache store failed");
+
+        // Case #1: no chunk at all, equivalent to having an inexistent
```

Comment:
I think I've seen more often `nonexistent` than `inexistent` in English, or "lacking"/"missing".